File: src (cpp)/animal/mammalia/wild_bunny.cpp
#include "wild_bunny.h"

WildBunny::WildBunny(int _x, int _y, int _weight) : def_ID(0),
                                                    def_ratio_meat(0),
                                                    def_ratio_plant(90) {
  ID = def_ID;
  position.SetX(_x);
  position.SetY(_y);
  ratio_meat = def_ratio_meat;
  ratio_plant = def_ratio_plant;
  is_land_animal = true;
  is_water_animal = false;
  is_air_animal = false;
  weight = _weight;
  for (int id = def_ID; id <= 21; id++) {
    AddEnemy(id);
  }
}
void WildBunny::Interact() {
  std::cout << "The wild bunny is looking at you with murderous intent" << std::endl;
}

File: src/virtualzoo/animal/mammalia/WildBunny.java
package virtualzoo.animal.mammalia;

import virtualzoo.animal.behavior.BehaviorWild;
import virtualzoo.animal.diet.Carnivora;

/**
 * Kelas WildBunny mendefinisikan atribut untuk wild bunny.
 *
 * @author Rionaldi Chandraseta - 13515077
 * @version 3.0
 * @since 3.0
 */
public class WildBunny extends Mammalia {

  /** <p>
   * Constructor
   *
   * Menciptakan wild bunny dengan posisi (locX, locY) dan berat animalWeight.
   * </p>
   * @param locX Nilai absis posisi.
   * @param locY Nilai ordinat posisi.
   * @param animalWeight Nilai berat wild bunny.
   */
  public WildBunny(int locX, int locY, int animalWeight) {
    super();
    animalBehavior = new BehaviorWild();
    animalDiet = new Carnivora(animalWeight, 90);
    id = 0;
    position.setLocation(locX, locY);
    isLandAnimal = true;
    isWaterAnimal = false;
    isAirAnimal = false;

    for (int id = 0; id < 22; id++) {
      animalBehavior.addEnemy(id);
    }
  }

  /**
   * Mengatur interaksi dengan wild bunny.
   */
  public void interact() {
    System.out.println("The wild bunny is looking at you with murderous intent");
  }
}
